ludwig/data/preprocessing.py

```diff
@@ -1671,88 +1679,65 @@ def _handle_missing_values(
-    feature_configs: List[FeatureConfigDict],
+def handle_features_with_prompt_config(
+    dataset_cols: Dict[str, Series],
     feature_name_to_preprocessing_parameters: Dict[str, PreprocessingConfigDict],
+    features: List[FeatureConfigDict],
     backend: Backend,
-) -> None:
-    """If output feature is a category feature and it's preprocessing has prompt_template, then we need to update
-    the input feature data with the prompt.
-
-    Example context:
-        In the example below, {review} is the input text feature, and {labels} are the labels that are passed
-        into the category output feature's preprocessing config.
-
-    Example input prompt:
-        Context information is below.
-        ###
-        {review}
-        ###
-        Given the context information and not prior knowledge, classify the context as one of: {labels}
-
-    Example output prompt:
-        Context information is below.
-        ###
-        The food at the restaurant was great!
-        ###
-        Given the context information and not prior knowledge, classify the context as one of: [positive, negative]
-    """
+    split_col: Optional[Series] = None,
+):
+    """Updates (in-place) dataset columns with prompt configurations containing a non-None task parameter.
 
-    # Recover input and output features from combined list of feature configs
-    input_features = []
-    output_features = []
-    for feature in feature_configs:
-        if ENCODER in feature:
-            input_features.append(feature)
-        elif DECODER in feature:
-            output_features.append(feature)
-
-    # If output feature is a category feature and it's preprocessing has prompt_template, then we need to
-    # add the prompt in the input text feature(s). This step assumes only one output feature.
-    names_of_features_to_substitute = None
-    vocab = None
-    prompt = None
-    for output_feature in output_features:
-        if output_feature[TYPE] != CATEGORY:
-            continue
-        if "prompt_template" not in feature_name_to_preprocessing_parameters[output_feature[NAME]]:
-            continue
+    Dataset columns that are updated here are enriched to have prompts as specified by the prompt configuration.
 
-        prompt: str = feature_name_to_preprocessing_parameters[output_feature[NAME]]["prompt_template"]
-        vocab: List = feature_name_to_preprocessing_parameters[output_feature[NAME]]["vocab"]
-        names_of_features_to_substitute: List = _extract_feature_names_from_prompt(prompt)
+    Args:
+        dataset_cols (Dict[str, Series]): Dataset columns.
+        feature_name_to_preprocessing_parameters (Dict[str, PreprocessingConfigDict]): Mapping from feature name to
+            preprocessing parameters.
+        features (List[FeatureConfigDict]): List of feature configurations.
+        df_engine (DataFrameEngine): Dataframe engine.
+        split_col (Optional[Series], optional): Split column. Defaults to None.
+    """
```

Comment:
Is it worth leaving an example of before and after enrichment in the doc-string like before?